renderdoc/driver/metal/metal_buffer.cpp

```diff
@@ -109,4 +109,32 @@ void WrappedMTLBuffer::didModifyRange(NS::Range &range)
   }
 }
 
+template <typename SerialiserType>
+bool WrappedMTLBuffer::Serialise_InternalModifyCPUContents(SerialiserType &ser, uint64_t start,
+                                                           uint64_t end)
+{
+  SERIALISE_ELEMENT_LOCAL(Buffer, this);
+  SERIALISE_ELEMENT(start).Important();
+  SERIALISE_ELEMENT(end).Important();
+  byte *pData = NULL;
+  uint64_t memSize = end - start;
+  if(ser.IsWriting())
+  {
+    pData = (byte *)Unwrap(this)->contents() + start;
+  }
+  if(IsReplayingAndReading())
+  {
+    pData = (byte *)Unwrap(Buffer)->contents() + start;
+  }
+
+  // serialise directly using buffer memory
+  ser.Serialise("data"_lit, pData, memSize, SerialiserFlags::NoFlags).Important();
```

Comment:
Minor nitpick, I would not mark the data itself as `Important()` as well as the range. The data can't be displayed in-line as anything but `x byte buffer` which is redundant with the start/end range. I think picking either one would be fine (either only the data, or only the range) - personally I would set the range as important since that is more directly useful and doesn't rely on the UI layer being able to gracefully display buffers in-line.

As a thought, it might be worth considering serialising start/size instead of start/end. Since this is an internal call anyway we have some freedom and that's more likely to be immediately intuitive to users.